### PR TITLE
Add filter command

### LINKED ISSUE #131: Add FilterCommand related parsing and appointment functionalities

### PR DESCRIPTION
**Pull request by yugan01**
Fixes #131

### CODE REVIEW

**Review by jolynleow2301 — APPROVED** [review]
LGTM! appointment can be shortened to "Appt" for standardisation purposes

File: src/main/java/seedu/address/logic/parser/FilterCommandParser.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        }
+        AppointmentDateFilter dateFilter = new AppointmentDateFilter(startDate, endDate, service);
+
+        return new FilterCommand(dateFilter);
+    }
+    /**
+     * Returns true if none of the prefixes contains empty {@code Optional} values in the given
+     * {@code ArgumentMultimap}.
+     */
+    private static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
```
Comment (jolynleow2301) [inline src/main/java/seedu/address/logic/parser/FilterCommandParser.java:55]: nice that you have created a new method here to support your function above

File: src/main/java/seedu/address/logic/parser/CliSyntax.java [hunk 2]
```diff
@@ -25,5 +25,7 @@ public class CliSyntax {
     public static final Prefix PREFIX_HEALTHRISK = new Prefix("rl/");
     public static final Prefix PREFIX_APPOINTMENT = new Prefix("appt/");
     public static final Prefix PREFIX_DATETIME = new Prefix("dt/");
+    public static final Prefix PREFIX_STARTDATE = new Prefix("sd/");
```
Comment (jolynleow2301) [inline src/main/java/seedu/address/logic/parser/CliSyntax.java:28]: can consider incorporating time as well

**Review by jolynleow2301 — COMMENTED** [review]
LGTM

**Review by DINESH1201 — COMMENTED** [review]
Good job on completing the filter command

**Pull request merged**